bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandWebSockets.java

```diff
@@ -0,0 +1,193 @@
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketConnect;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketError;
+import org.eclipse.jetty.websocket.api.annotations.OnWebSocketMessage;
+import org.eclipse.jetty.websocket.api.annotations.WebSocket;
+import org.eclipse.jetty.websocket.client.ClientUpgradeRequest;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.openhab.binding.touchwand.internal.dto.TouchWandUnitData;
+import org.openhab.binding.touchwand.internal.dto.TouchWandUnitFromJson;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link TouchWandWebSockets} class implements WebSockets API to TouchWand controller
+ *
+ * @author Roie Geron - Initial contribution
+ */
+@NonNullByDefault
+public class TouchWandWebSockets {
+
+    private static final int CONNECT_TIMEOUT_SEC = 10;
+    private static final int WEBSOCKET_RECONNECT_INTERVAL_SEC = CONNECT_TIMEOUT_SEC * 2;
+    private static final int WEBSOCKET_IDLE_TIMEOUT_MS = CONNECT_TIMEOUT_SEC * 10 * 1000;
+    private final Logger logger = LoggerFactory.getLogger(TouchWandWebSockets.class);
+    private static final String WS_ENDPOINT_TOUCHWAND = "/async";
+
+    private WebSocketClient client;
+    private String controllerAddress;
+    private TouchWandSocket touchWandSocket;
+    private boolean isShutDown = false;
+    private List<TouchWandUnitStatusUpdateListener> listeners = new ArrayList<>();
+    private @Nullable ScheduledFuture<?> socketReconnect;
+    private @Nullable URI uri;
+
+    ScheduledExecutorService scheduler;
+
+    public TouchWandWebSockets(String ipAddress, ScheduledExecutorService scheduler) {
+        client = new WebSocketClient();
+        touchWandSocket = new TouchWandSocket();
+        this.controllerAddress = ipAddress;
+        this.scheduler = scheduler;
+        socketReconnect = null;
+    }
+
+    public void connect() {
+        try {
+            uri = new URI("ws://" + controllerAddress + WS_ENDPOINT_TOUCHWAND);
+        } catch (URISyntaxException e) {
+            logger.warn("URI not valid {} message {}", uri, e.getMessage());
+            return;
+        }
+
+        client.setConnectTimeout(CONNECT_TIMEOUT_SEC);
+        ClientUpgradeRequest request = new ClientUpgradeRequest();
+        request.setSubProtocols("relay_protocol");
+
+        try {
+            client.start();
+            client.connect(touchWandSocket, uri, request);
+        } catch (Exception e) {
+            logger.warn("Could not connect webSocket URI {} message {}", uri, e.getMessage());
+            return;
+        }
+    }
+
+    public void dispose() {
+        isShutDown = true;
+        try {
+            client.stop();
+        } catch (Exception e) {
+            logger.warn("Could not stop webSocketClient,  message {}", e.getMessage());
+        }
```

Comment:
Shouldn't the `socketReconnect` be canceled here?

Reply:
indeed it should....
owe you a beer....